bindings/pydeck/pydeck/bindings/layer.py

```diff
@@ -89,35 +90,79 @@ def __init__(
                     # Skip quoted strings
-                    kwargs[k] = v.replace(v[0], '')
+                    kwargs[k] = v.replace(v[0], "")
                 elif isinstance(v, str):
                     # Have @deck.gl/json treat strings values as functions
                     kwargs[k] = FUNCTION_IDENTIFIER + v
 
                 elif isinstance(v, list) and v != [] and isinstance(v[0], str):
                     # Allows the user to pass lists e.g. to specify coordinates
-                    array_as_str = ''
+                    array_as_str = ""
                     for i, identifier in enumerate(v):
                         if i == len(v) - 1:
-                            array_as_str += '{}'.format(identifier)
+                            array_as_str += "{}".format(identifier)
                         else:
-                            array_as_str += '{}, '.format(identifier)
-                    kwargs[k] = '{}[{}]'.format(FUNCTION_IDENTIFIER, array_as_str)
+                            array_as_str += "{}, ".format(identifier)
+                    kwargs[k] = "{}[{}]".format(FUNCTION_IDENTIFIER, array_as_str)
 
             self.__dict__.update(kwargs)
 
+        self._data = None
+        self.use_binary_transport = use_binary_transport
+        self._binary_data = None
+        if not self.use_binary_transport:
+            self.data = data.to_dict(orient="records") if is_pandas_df(data) else data
+        else:
+            self.data = data
+
     @property
     def data(self):
         return self._data
 
     @data.setter
     def data(self, data_set):
-        """Make the data attribute a list no matter the input type"""
-        if is_pandas_df(data_set):
-            self._data = data_set.to_dict(orient='records')
+        """Make the data attribute a list no matter the input type, unless
+        use_binary_transport is specified, which case we circumvent
+        serializing the data to JSON
+        """
+        if self.use_binary_transport:
+            self._binary_data = self._prepare_binary_data(data_set)
+        elif is_pandas_df(data_set):
+            self._data = data_set.to_dict(orient="records")
         else:
             self._data = data_set
 
+    def get_binary_data(self):
+        if not self.use_binary_transport:
+            raise BinaryTransportException(
+                "Layer must be flagged with `use_binary_transport=True`"
+            )
+        return self._binary_data
+
+    def _prepare_binary_data(self, data_set):
+        # Binary format conversion gives a sizable speedup but requires
+        # slightly stricter standards for data input
+        if not is_pandas_df(data_set):
+            raise BinaryTransportException(
+                "Layer data must be a `pandas.DataFrame` type"
+            )
+
+        layer_accessors = self._kwargs
+        inv_map = {v: k for k, v in layer_accessors.items()}
+
+        blobs = []
+        for column in data_set.columns:
```

Comment:
What happens if the JSON references something in the binary data that doesn't exist?  I was looking for a test case for this path but I don't see it.